rasa/shared/core/trackers.py

```diff
@@ -624,12 +625,13 @@ def update(self, event: Event, domain: Optional[Domain] = None) -> None:
         event.apply_to(self)
 
         if domain and isinstance(event, (UserUttered, DefinePrevUserUtteredEntities)):
-            entities = event.entities
             if isinstance(event, UserUttered):
                 # Rather get entities from `parse_data` as
                 # `DefinePrevUserUtteredEntities` might have already affected the
                 # `UserUttered.entities` attribute
-                entities = event.parse_data["entities"]
+                entities = event.parse_data[ENTITIES]
+            else:
+                entities = event.entities
```

Comment:
what about:
```
if domain:
   if isinstance(event, UserUttered):
   if isinstance(event, DefinePrevUserUtteredEntities):
```
?

Reply:
I think that's style dependent and should be optional to address.

Reply:
I mean... this is colossal `if domain and isinstance(event, (UserUttered, DefinePrevUserUtteredEntities)):` and then the solitary `else` is confusing

Reply:
it takes more than a second to understand this `if` for someone else

Reply:
I don't think it is style dependent, we should make sure that the code is as simple as possible to grasp by looking at it

Reply:
I still think it's clear. Happy to hear @joejuzl 's opinion on it.